tensorflow/lite/micro/tools/make/targets/cortex_m_generic_makefile.inc

```diff
@@ -1,156 +1,145 @@
+# soft-abi=soft disables MVE - use softfp instead for M55.
+ifneq ($(filter cortex-m55%,$(TARGET_ARCH)),)
+  ifeq ($(FLOAT),soft)
+    FLOAT=softfp
   endif
+endif
 
-  # Toolchain specfic flags
-  ifeq ($(filter armclang,$(ALL_TAGS)),armclang)
-    CXX_TOOL  := armclang
-    CC_TOOL   := armclang
-    AR_TOOL   := armar
-    LD        := armlink
-    FLAGS_ARMC = \
-      --target=arm-arm-none-eabi \
-      -mcpu=$(TARGET_ARCH)
-    ifeq ($(BUILD_TYPE), release)
-       FLAGS_ARMC += -Wno-unused-private-field
-    endif
+# Toolchain specfic flags
+ifeq ($(TOOLCHAIN), armclang))
+  CXX_TOOL  := armclang
+  CC_TOOL   := armclang
+  AR_TOOL   := armar
+  LD    := armlink
+  FLAGS_ARMC = \
+    --target=arm-arm-none-eabi \
+    -mcpu=$(TARGET_ARCH)
 
-    CXXFLAGS += $(FLAGS_ARMC)
-    CCFLAGS += $(FLAGS_ARMC)
-    LDFLAGS += $(ARM_LDFLAGS)
+  CXXFLAGS += $(FLAGS_ARMC)
+  CCFLAGS += $(FLAGS_ARMC)
+  LDFLAGS += $(ARM_LDFLAGS)
 
-    # Arm Compiler will not link the Math library (see below), therefore we're filtering it out.
-    # See Fatal error: L6450U: Cannot find library m:
-    # "Arm Compiler is designed to run in a bare metal environment,
-    # and automatically includes implementations of these functions,
-    # and so no such flag is necessary."
-    # https://developer.arm.com/documentation/100891/0611/troubleshooting/general-troubleshooting-advice
-    MICROLITE_LIBS := $(filter-out -lm,$(MICROLITE_LIBS))
-  else ifeq ($(filter armgcc,$(ALL_TAGS)),armgcc)
-    ifeq ($(shell which arm-none-eabi-gcc),)
-      export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
-      $(eval $(call add_third_party_download,$(GCC_EMBEDDED_URL),$(GCC_EMBEDDED_MD5),gcc_embedded,))
-    endif
+  # Arm Compiler will not link the Math library (see below), therefore we're filtering it out.
+  # See Fatal error: L6450U: Cannot find library m:
+  # "Arm Compiler is designed to run in a bare metal environment,
+  # and automatically includes implementations of these functions,
+  # and so no such flag is necessary."
+  # https://developer.arm.com/documentation/100891/0611/troubleshooting/general-troubleshooting-advice
+  MICROLITE_LIBS := $(filter-out -lm,$(MICROLITE_LIBS))
+else ifeq ($(TOOLCHAIN), armgcc))
+  ifeq ($(shell which arm-none-eabi-gcc),)
+    export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
+    $(eval $(call add_third_party_download,$(GCC_EMBEDDED_URL),$(GCC_EMBEDDED_MD5),gcc_embedded,))
+  endif
 
-    CXX_TOOL  := arm-none-eabi-gcc
-    CC_TOOL   := arm-none-eabi-gcc
-    AR_TOOL   := arm-none-eabi-gcc-ar
-    LD        := arm-none-eabi-ld
-    ifneq ($(filter cortex-m55%,$(GCC_TARGET_ARCH)),)
-      $(error Micro architecure support is not available yet for $(GCC_TARGET_ARCH))
-    else
-      FLAGS_GCC = -mcpu=$(GCC_TARGET_ARCH)
-    endif
-    CXXFLAGS += $(FLAGS_GCC)
-    CCFLAGS += $(FLAGS_GCC)
-    LDFLAGS +=
+  TARGET_TOOLCHAIN_PREFIX := arm-none-eabi-
+  
+  ifneq ($(filter cortex-m55%,$(TARGET_ARCH)),)
```

Comment:
use findstring instead?

```make
ifeq (cortex-m55, $(findstring cortex-m55, $(TARGET_ARCH)))
```

Reply:
This is trying to find all cortex-m55, e.g. cortex-m55+nofp, findstring won't work with pattern '%'.